### PR TITLE
Add additional picard iterations

### PR DESCRIPTION
**Pull request by marchdf**
companion PR to https://github.com/Exawind/nalu-wind/pull/940

### CODE REVIEW

File: src/NaluWind.cpp [hunk 1]
```diff
@@ -83,6 +83,12 @@ void NaluWind::advance_timestep()
         realm->advance_time_step();
 }
 
+void NaluWind::nonlinear_iteration(const int n)
```
Comment (michaeljbrazell) [inline src/NaluWind.cpp:86]: I think this is what you meant earlier, there is an inconsistency in the naming (nonlinear_iteration vs nonlinear_iterations). Also, in the future maybe we should add a for loop here? And have two parameters? One that controls frequency of exchange solution between codes (now fixed at 1) and another that runs a fixed number of picards (like what we have with the input nonlinear_its)

Comment (ashesh2512) [reply src/NaluWind.cpp:86]: we should make it consistent!

Comment (marchdf) [reply src/NaluWind.cpp]: Ok it's consistent ;) 

Do we want to add the second piece of Mike's comments now? Or wait?

**Comment by marchdf** [comment]
Ok I think this is good to go... except that #19 broke something so this is based on `acae22c`

**Pull request merged**